x-pack/plugins/observability/public/components/shared/alert_search_bar/services.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useContext, useMemo } from 'react';
+import { ObservabilityAlertSearchBarDependencies, Services } from './types';
+
+const ObservabilityAlertSearchBarContext = React.createContext<Services | null>(null);
+
+export const ObservabilityAlertSearchBarProvider: FC<ObservabilityAlertSearchBarDependencies> = ({
+  children,
+  data: {
+    query: {
+      timefilter: { timefilter: timeFilterService },
+    },
+  },
+  useToasts,
```

Comment:
I've tried to pass toast directly similar to other dependencies but it will cause an issue regarding `this` in the toasts API. So I passed `useToasts` to call `useKibana` when we need to show the toast. I saw `useToasts` in other plugins such as `cases` too.